+Base/GitPanel.m

```diff
@@ -0,0 +1,164 @@
+        end
+        function str = info(obj)
+            thisbranch = makeHTML(obj.thisbranch());
+            
+            words_ = split(thisbranch, ' ');
+            words = {};
+            
+            jj = 1;
+            for ii = 1:length(words_)
+                if ~isempty(words_{ii})
+                    words{jj} = words_{ii}; %#ok<AGROW>
+                    jj = jj + 1;
+                end
+            end
+            
+            message_ = split(thisbranch, words{2});
+            messageraw = strtrim(message_{end});
+            
+            brackets = split(messageraw, {'[', ']'});
+            
+            while ~isempty(brackets) && isempty(brackets{1})
+                brackets(1) = [];
+            end
+            
+            message = '';
+
+            third = words{3};
+            
+            test = '';
+            if numel(third) >= 6
+                test = third(1:6);
+            end
+
+            switch test
+                case '[ahead'
+                    commas = split(brackets{1}, {', '});
+                    
+                    if numel(commas) == 2
+                        message = ['&nbsp;&nbsp;<font color="orange">[' commas{1} '</font>, <font color="red">' commas{2} ']</font>'];% messageraw((numel(brackets{1})+3):end)];
+                    
+                    else
+                        message = ['&nbsp;&nbsp;<font color="orange">[' brackets{1} ']</font>'];% messageraw((numel(brackets{1})+3):end)];
+                    end
+                case '[behin'
+                    message = ['&nbsp;&nbsp;<font color="red">[' brackets{1} ']</font>'];% messageraw((numel(brackets{1})+3):end)];
+                otherwise
+            end
+            
+            status = git('status');
+            
+            modified = contains(status,'Changes not staged for commit:') || contains(status,'Changes to be committed:');
+            untracked = contains(status,'Untracked files:');
+            
+            if modified && untracked
+                message = [message '&nbsp;&nbsp;<I><font color=rgb(255,69,0)>Modified and <font color="red">Untracked</font> Files</font></I>'];
+            elseif modified
+                message = [message '&nbsp;&nbsp;<font color=rgb(255,69,0)><I>Modified Files</I></font>'];
+            elseif untracked
+                message = [message '&nbsp;&nbsp;<font color="red"><I>Untracked Files</I></font>'];
+            end
+            
+            str = ['<html><font color="blue"><B>' words{1} '</B>&nbsp;&nbsp;<I>' words{2} '</I></font>' message];
+        end
+        function str = tooltip(obj)
+            str_ = strrep(git('status --ahead-behind --show-stash'), '/', ' / ');
+            str__ = split(str_, newline);
+            
+            assert(~isempty(str__));
+            
+            str = ['On branch ' obj.thisbranch() newline str_((numel(str__{1})+1):end)];
+        end
+    
+    end
+end
+
+function str = makeHTML(str)    % Not rigorous
```

Comment:
Any reason not to make this a more rigorous helper function? Even if it isn't fully comprehensive, it could be a start to one that we add to.

Reply:
Reason: laziness.